react/react1/week3/to-do/src/components/FetchData.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import "react-datepicker/dist/react-datepicker.css";
+import Header from "./Header";
+import TodoItem from "./ToDoItem";
+import DatePicker from "react-datepicker";
+const FetchData = () => {
+  const [todos, setToDos] = useState([]);
+  const [edit, setEEdit] = useState(false);
+  const [newToDo, setNewToDo] = useState({});
+  const [deadLine, setDeadLine] = useState(new Date());
+  const NewToDo = (description) => {
+    setNewToDo({
+      id: Math.floor(Math.random() * (100 - 4 + 1) + 4),
+      description: description,
+      deadline: deadLine,
+      done: false,
+    });
+  };
+  const AddToDo = () => {
+    setToDos((previous) => [...previous, newToDo]);
+  };
+  const DeleteToDo = (id) => {
+    setToDos(todos.filter((eachToDo) => eachToDo.id !== id));
+  };
+  const editMode = (id) => {
+    setEEdit(!edit);
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+  };
+  const handleDesc = (desc, id) => {
```

Comment:
Do you mean `handleDesc ` is handling the change of the description of a todo?
Names should be self-descriptive. Use can something like `EditDescription` or `HandleDescrption`.
Always use full words not short form or abbreviated unless they are globally defined.

Reply:
I will keep this in mind 👍🏻and thanks a lot for your detail review 🙏🏾.